### PR TITLE
Adding user verification support

### PR DESCRIPTION
**Pull request by chenrozenes**

### CODE REVIEW

File: src/methods.js [hunk 1]
```diff
@@ -72,5 +72,48 @@ export default {
...
+  /**
+   * Compares the token provided to the verification token that is saved on the user
+   * Returns a promise - resolves if the tokens are equal, otherwise rejects and throws an error
+   * @param token
+   * @returns {Promise}
+   */
+  verifyVerificationToken(token) {
+    return new Promise((resolve, reject) => {
+      if (this.verificationToken === token) {
+        resolve();
```
Comment (rom10) [inline src/methods.js]: It's better to return something, maybe true

File: src/statics.js [hunk 2]
```diff
@@ -11,9 +11,10 @@ export default {
    * @returns {Promise.<T>}
    */
   login(username, password) {
-    return this.findOne({username: username}).select('services.password.bcrypt').exec()
+    // Selecting password and emails (for getting the isVerified virtual)
+    return this.findOne({username: username}).select('services.password.bcrypt emails').execAsync()
     .then((user) => {
-      if (!!user) {
+      if (!!user && user.isVerified === true) {
```
Comment (rom10) [inline src/statics.js]: Use a config, is someone uses this package, he won't be able to login

File: README.md [hunk 3]
```diff
@@ -31,14 +31,17 @@ export default mongoose.model('User', UserSchema);
 ```
 
 <b>Statics</b>:
-- User.login(username, password) - Return Promise of the user if success, false if not
+- User.login(username, password) - Return Promise of the user if success, false if not. The user must be verified in order to login.
```
Comment (asafdav) [inline README.md]: as @rom10 mentioned you can add a third parameter verify or something like that

File: src/methods.js [hunk 4]
```diff
@@ -72,5 +72,48 @@ export default {
...
+  },
+
+  /**
+   * Generates a token for verifying a new user.
+   * Later the token can be verified with "verifyVerificationToken"
+   * @returns {*}
+   */
+  generateVerificationToken() {
+    const token = chance.hash(); // 40 chars by default
+    this.verificationToken = token;
```
Comment (asafdav) [inline src/methods.js]: I think that in Meteor it saves the verificationToken under the email and not in the main object

File: src/user.js [hunk 5]
```diff
@@ -48,11 +48,29 @@ export default function UserPlugin(schema, options) {
...
    * Add Virtual fot email (only get!)
-   * user.emails[primary].address -> user.email
    */
   schema.virtual('email').get(function () {
-    const email = _.find(this.emails, {primary: true});
-    return email ? email.address : undefined;
+    return _.get(this.emails, '[0].address');
+  });
+
+  schema.virtual('verificationToken').set(function(token) {
```
Comment (asafdav) [inline src/user.js]: super cool !

**Pull request merged**